src/components/Popup/Popup.tsx

```diff
@@ -80,78 +85,61 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
 
     /** Element to be rendered in-place where the popup is defined. */
     trigger: PropTypes.node,
-
-    /** Additional CSS styles to apply to the component instance.  */
-    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-
-    /** Override for theme site variables to allow modifications of component styling via themes. */
-    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-
-    /** Accessibility behavior if overridden by the user. */
-    accessibility: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
   public static defaultProps: IPopupProps = {
-    as: Portal,
+    accessibility: PopupBehavior,
     align: 'start',
     position: 'above',
-    accessibility: PopupBehavior,
   }
 
-  public state = { triggerRef: undefined, popupOpened: false }
+  public static autoControlledProps = ['open']
 
   protected actionHandlers: AccessibilityActionHandlers = {
-    performClick: e => this.onTriggerClick(e),
-    closeAndFocusTrigger: e => this.handlePopupState(e, () => false, this.focusTrigger),
+    toggle: e => _.invoke(this.props, 'onOpenChange', e, !this.props.open),
+    closeAndFocusTrigger: e => {
+      _.invoke(this.props, 'onOpenChange', false)
+      _.invoke(this.state.triggerRef, 'focus')
+    },
   }
 
+  public state = { triggerRef: undefined }
+
   public renderComponent({
-    ElementType,
-    classes,
-    accessibility,
-    rest,
     rtl,
+    accessibility,
   }: IRenderResultConfig<IPopupProps>): React.ReactNode {
-    const { children, trigger, position, align } = this.props
+    const { children, trigger } = this.props
 
     return (
-      <ElementType
-        className={classes.root}
-        {...rest}
-        open={this.state.popupOpened}
-        trigger={trigger}
-        triggerRef={this.handleTriggerRef}
-        triggerAccessibility={
-          {
-            attributes: accessibility.attributes.trigger,
-            keyHandlers: accessibility.keyHandlers.trigger,
-          } as TriggerAccessibility
-        }
-        onOutsideClick={e => this.handlePopupState(e, () => false)}
-        onTriggerClick={this.onTriggerClick}
-      >
-        {childrenExist(children)
-          ? children
-          : this.renderContent(
-              computePopupPlacement({ align, position, rtl }),
-              this.renderPopperChildren.bind(this, rtl, accessibility),
-            )}
-      </ElementType>
+      <>
+        <Ref
+          innerRef={domNode => {
+            this.setState({ triggerRef: domNode })
+          }}
```

Comment:
we should define a method for this and not define it here every time since it's hurting perf and it's not a good React practice

Reply:
here is the excerpt from official docs :) 

https://reactjs.org/docs/faq-functions.html

![image](https://user-images.githubusercontent.com/9024564/46141374-09aa5780-c254-11e8-8a03-3338b2b00b6a.png)

And now it seems that we cannot claim that there are any perf issues because of that. I will change this, but it seems that it is better to obtain measurements first, to avoid premature optimisations with unclear benefit.